src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/ProjectsController.java

```diff
@@ -428,7 +428,7 @@ public String getProjectSamplesMetadataUploadPage(final Model model, @PathVariab
 	 * @param principal {@link Principal} currently logged in use
 	 * @return {@link String} the path to the metadata import page
 	 */
-	@RequestMapping(value = "/projects/{projectId}/sample-metadata/upload2", method = RequestMethod.GET)
+	@RequestMapping(value = "/projects/{projectId}/sample-metadata/upload2/file", method = RequestMethod.GET)
```

Comment:
We've been trying to get away from the `@RequestMapping` stuff within controllers in our recent refactors. Replacing them with `@GetMapping`, `@PostMapping`, etc depending on the type of request. This could be simplified down to:

`@GetMapping("/projects/{projectId}/sample-metadata/upload2/file")`. Also instead of using PathVariable we have been switching over to using RequestParam for method params:

`public String getProjectSamplesMetadataUploadPage2(final Model model, @PathVariable long projectId,
			Principal principal)`
would become:

`public String getProjectSamplesMetadataUploadPage2(final Model model, @RequestParam long projectId,
			Principal principal)`
			
and when making the call from the api instead of passing the projectId as a part of the URL you would add it to the end of the url like this:

http://localhost:8080/projects/sample-metadata/upload2/file`?projectId=${projectId}`

I hope this makes sense. We can discuss it Monday

Reply:
The @RequestMapping stuff also applies to the controller below

Reply:
Please see 26d0d3fea80f3ef0b5a5ec0b44d6aa43cd5b7f60. I just don't know if there's a better way to pass the projectId to both SampleMetadatImportMapHeaders.jsx and SampleMetadataImportUploadFile.jsx.